libs/utils/src/lsn.rs

```diff
@@ -13,10 +13,77 @@ use crate::seqwait::MonotonicCounter;
 pub const XLOG_BLCKSZ: u32 = 8192;
 
 /// A Postgres LSN (Log Sequence Number), also known as an XLogRecPtr
-#[derive(Clone, Copy, Eq, Ord, PartialEq, PartialOrd, Hash, Serialize, Deserialize)]
-#[serde(transparent)]
+#[derive(Clone, Copy, Eq, Ord, PartialEq, PartialOrd, Hash)]
 pub struct Lsn(pub u64);
 
+impl Serialize for Lsn {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        if serializer.is_human_readable() {
+            format!("{:X}/{:X}", self.0 >> 32, self.0 & 0xffffffff).serialize(serializer)
```

Comment:
```suggestion
            serializer.collect_str(self)
```

This will use the Display impl.

Reply:
Done https://github.com/neondatabase/neon/pull/5335/commits/8251563491214e845bff4343e16958e18b6d59f1